subsys/console/getchar.c

```diff
@@ -46,12 +47,29 @@ int console_getchar(void)
 	return c;
 }
 
-void console_init(void)
+int console_init(void)
 {
 	struct device *uart_dev;
+	int ret;
 
 	uart_dev = device_get_binding(CONFIG_UART_CONSOLE_ON_DEV_NAME);
-	tty_init(&console_serial, uart_dev);
+	ret = tty_init(&console_serial, uart_dev);
+
+	if (ret) {
+		return ret;
+	}
+
+	/* Checks device driver supports for interrupt driven data transfers. */
+	if (CONFIG_CONSOLE_GETCHAR_BUFSIZE + CONFIG_CONSOLE_PUTCHAR_BUFSIZE) {
+		const struct uart_driver_api *api =
+			(const struct uart_driver_api *)uart_dev->driver_api;
+		if (!api->irq_callback_set) {
+			return -EINVAL;
```

Comment:
EINVAL means "invalid value was passed as parameter". As nothing was passed in here, it's not a good error code. Please look up something like ENOTSUPP.

Reply:
Agree on that, updated.